planner/core/task.go

```diff
@@ -930,60 +935,76 @@ func (p *basePhysicalAgg) newPartialAggregate(copToFlash bool) (partial, final P
 		finalAggFunc.RetTp = aggFunc.RetTp
 		finalAggFuncs[i] = finalAggFunc
 	}
-
 	// add group by columns
-	groupByItems := make([]expression.Expression, 0, len(p.GroupByItems))
-	for _, gbyExpr := range p.GroupByItems {
+	finalGbyItems = make([]expression.Expression, 0, len(groupByItems))
+	for _, gbyExpr := range groupByItems {
 		gbyCol := &expression.Column{
-			UniqueID: p.ctx.GetSessionVars().AllocPlanColumnID(),
+			UniqueID: sctx.GetSessionVars().AllocPlanColumnID(),
 			RetType:  gbyExpr.GetType(),
 		}
 		partialSchema.Append(gbyCol)
-		groupByItems = append(groupByItems, gbyCol)
+		finalGbyItems = append(finalGbyItems, gbyCol)
 	}
+	return
```

Comment:
Should we set `Mode` of `finalAggFuncs` to `FinalMode` in this function?

Reply:
Yes, we have done that in this function.

Reply:
Yes, and we have done that in this function.